src/app/info/ClientSideHelpers.tsx

```diff
@@ -5,29 +5,30 @@ import { useState } from "react";
 import DefaultWikiItemView from "@/app/info/DefaultWikiItemView";
 import EditWikiItemView from "@/app/info/EditWikiItemView";
 
-export const enterEditMode = (setEditMode: (editMode: boolean) => void): void => {
-    setEditMode(true);
+export const enterEditMode = (setIsInEditMode: (editMode: boolean) => void): void => {
+    setIsInEditMode(true);
 };
 
 interface EditProps {
-    editMode: boolean;
+    isInEditMode: boolean;
     row?: DbWikiRow;
 }
 
-export const EditModeDependentItem: React.FC<EditProps> = ({ editMode, row }) => {
+export const EditModeDependentItem: React.FC<EditProps> = ({ isInEditMode: isInEditMode, row }) => {
     const [rowRef, setRowRef] = useState<DbWikiRow | undefined>(row);
-    const [editModeRef, setEditModeRef] = useState(editMode);
+    const [isInEditModeRef, setIsInEditModeRef] = useState(isInEditMode);
     return (
         <>
-            {!editModeRef ? (
-                <DefaultWikiItemView rowRef={rowRef} setEditModeRef={setEditModeRef} />
-            ) : (
-                <EditWikiItemView
-                    rowRef={rowRef}
-                    setRowRef={setRowRef}
-                    setEditModeRef={setEditModeRef}
-                />
-            )}
+            {rowRef &&
```

Comment:
would we be able to move the rowRef condition even further out? i don't think EditModeDependentItem needs to be called if row is undefined

Reply:
rowRef can only become undefined once deleted in EditWikiItemView. This is the highest level client side file in the info route and we need to store rowRef as a ref so that it can be dynamically updated (deleted). We cannot store this ref any higher as that would require storing it on a SSR page which does not support the useState hook.